logos/src/lexer.rs

```diff
@@ -2,14 +2,14 @@ use std::ops::Range;
 
 use super::internal::LexerInternal;
 use super::Logos;
-use crate::source::{self, Source, WithSource};
+use crate::source::{self, Source};
 
 /// `Lexer` is the main struct of the crate that allows you to read through a
 /// `Source` and produce tokens for enums implementing the `Logos` trait.
 #[derive(Clone)]
-pub struct Lexer<Token: Logos, Source> {
+pub struct Lexer<'source, Token: Logos, Source: ?Sized> {
     /// Source from which the Lexer is reading tokens.
-    pub source: Source,
+    pub source: &'source Source,
```

Comment:
So, I'm no good at traits/deriving/etc, but
I believe the reason I no longer see an impl ToOwned, has to do with this change to Lexer.

in [ToOwned](https://doc.rust-lang.org/std/borrow/trait.ToOwned.html) there is an `impl ToOwned<T> for T where T: Clone`,
`&str` is Clone while `str` is not Clone, merely ToOwned.

So somehow this results in Lexer<'source, str> not being clone?

Reply:
Ah, right. The `Clone` derive checks that all generic types are `Clone`, it's not smart enough to realize that `Source` is behind a reference which makes it not only `Clone` but `Copy`.

I've added a manual `Clone` impl for `Lexer`, so that should be fine now.

Reply:
Thanks, the testcase which caught that is [here](https://github.com/ratmice/logos/commit/f56f1762fce7cf667bf5e4a3c6ed180b608f79cc) and it works now, I had to move it lexer-mode-iterator branch, bring it in if you'd like.

Reply:
Cool!

There isn't much there that's related to `logos`/`logos-derive`, but it does at least test that lexer is `Clone`, so why not :P.